### PR TITLE
Fix focus and camera handling for newer iOS devices with triple cameras (+ other fixes)

### PR DESCRIPTION
**Pull request by michaelwolz**
## Overview

This PR fixes several open issues, particularly on iOS, by addressing problems related to newer iOS devices with triple cameras (iPhone 14 Pro and later). Also this PR cleans up the iOS code, removes dead code and updates deprecated function calls that triggered warnings.

We are currently using this adapted version of the plugin in our app and it works perfectly fine for us. I don't know if this plugin is still actively maintained or not but I appreciate any feedback and am also willing to do adaptions if this is necessary to get it merged into the official codebase. 

## Changes

- Switched to using the virtual [builtInTripleCamera](https://developer.apple.com/documentation/avfoundation/avcapturedevice/devicetype/3377621-builtintriplecamera), which automatically switches between the three lenses based on the focused object.
- Set the initial zoom factor to 2 for the builtInTripleCamera to avoid the default use of the ultra-wide-angle lens, which can cause a "fisheye" effect. This zoom factor matches the "1x" setting of the native camera app.
- Cleaned up camera initialization logic by restructuring the code (this was done to improve performance when starting the capture session)
- Make high resolution mode work on iOS by using dedicated [AVCaptureSession preset `photo`](https://developer.apple.com/documentation/avfoundation/avcapturesession/preset/1390112-photo)
- Throw an error when calling `startRecordVideo()` or `stopRecordVideo()` on iOS and adapted the readme accordingly. This was not implemented even though documentation says it is. I think it's better to fully remove this from the code and refer to another solution such as: https://github.com/capacitor-community/video-recorder. I'd wish I would have a better solution but since I am not using this feature I decided to just change the current behvior to not confuse devs that call these methods and expect any output from it (also unnecessarily adding the `DataOutput` for the video capture was slowing down starting the camera by a lot). 
- Placed the camera preview overlay on the correct position when using `x` and `y` coordinates
- Added `checkPermissions` and `requestPermissions` method for iOS
- Fixed rotation logic of images on Android
- Added some documentation to the code (iOS)

## Resolved Issues

- Fixes: capacitor-community/camera-preview#267
- Fixes: capacitor-community/camera-preview#263
- Fixes: capacitor-community/camera-preview#248
- Fixes: capacitor-community/camera-preview#218
- Fixes: capacitor-community/camera-preview#324
- Fixes: capacitor-community/camera-preview#110

**Note**: I just saw that #339 also updates to the new Camera v2 API which would be even better due to *many* deprecation warnings with the old code but still this PR will fix the rotation logic with the old API

### CODE REVIEW

File: ios/Plugin/UIImage.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
+extension UIImage {
```
Comment (michaelwolz) [inline ios/Plugin/UIImage.swift:1]: This was just moved to a separate file. Was previously located in `CameraController.swift`

**Comment by florianleger** [comment]
It seems that the captureSample method does not work (when I call it, I never get the response or error).
Otherwise, the focus fix works just fine!

**Comment by michaelwolz** [comment]
@florianleger hm, yes probably [due to the removal of the [AVCaptureVideoDataOutput](https://github.com/capacitor-community/camera-preview/pull/342/files#diff-1063fd678ec7ace4faa94761cf006ee3d8077e4462cfec553fc82748b7d8427fL116) which I did because of severe performance problems with handling multiple outputs. 

In theory we might make this an optional feature which can be enabled via some flag and then configures a DataOutput again :thinking:

**Comment by michaelwolz** [comment]
Since I did quite a lot of opinionated changes to the code I decided to publish a lite version of the plugin on my own until further development goes on with the official version :).

Feel free to use my version as long as you just want to take photos and do not require video recordings etc: https://github.com/michaelwolz/camera-preview-lite

**Comment by selected-pixel-jameson** [comment]
@michaelwolz THANK YOU!!!!

**Pull request closed without merging**